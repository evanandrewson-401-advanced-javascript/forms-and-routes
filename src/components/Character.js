import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ name, photoUrl }) => {
  return (
    <div>
      <img src={photoUrl} />
      <h1>{name}</h1>
    </div>
  );
};

Character.propTypes = {
  name: PropTypes.string.isRequired,
  photoUrl: PropTypes.string.isRequired
};

export default Character;
