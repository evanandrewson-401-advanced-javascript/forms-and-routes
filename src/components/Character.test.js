import Character from './Character';
import React from 'react';
import { shallow } from 'enzyme';

const character = {
  _id: '5cf5679a915ecad153ab6a45',
  allies: [
    'Hakoda'
  ],
  enemies: [
    'Ozai'
  ],
  photoUrl: 'https://vignette.wikia.nocookie.net/avatar/images/c/cc/Sokka.png/revision/latest?cb=20140905085428',
  name: 'Sokka',
  gender: 'Male',
  eye: 'Blue',
  hair: 'Dark brown ',
  skin: 'Brown',
  love: ' Kaya (love interest; formerly) Suki (girlfriend) Yue (girlfriend; deceased; reincarnated as the moon spirit)',
  weapon: 'Sokka\'s weapons',
  profession: ' Politician Tactician Warrior',
  position: ' Chief of Southern Water Tribe (formerly) Map reader Son of Head Chieftain Hakoda Southern Water Tribe representative and Chairman of the United Republic Council (formerly)',
  affiliation: ' Bendless Boomerang Club[7] Southern Water Tribe Team Avatar United Republic Council',
  first: '',
  __v: 0,
};

const toDetailPage = () => {};

describe('Character', () => {
  it('renders a Character', () => {
    const wrapper = shallow(<Character {...character} toDetailPage={toDetailPage} />);
    expect(wrapper).toMatchSnapshot();
  });
});
