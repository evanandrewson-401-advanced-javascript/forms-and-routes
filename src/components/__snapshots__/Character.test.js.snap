// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character renders a Character 1`] = `
<button
  onClick={[Function]}
  value="5cf5679a915ecad153ab6a45"
>
  <img
    src="https://vignette.wikia.nocookie.net/avatar/images/c/cc/Sokka.png/revision/latest?cb=20140905085428"
  />
  <h1>
    Sokka
  </h1>
</button>
`;
